src/v/archival/service.cc

```diff
@@ -180,23 +187,26 @@ scheduler_service_impl::scheduler_service_impl(
   : scheduler_service_impl(config.local(), remote, api, pm, tt) {}
 
 void scheduler_service_impl::rearm_timer() {
-    (void)ss::with_gate(_gate, [this] {
-        return reconcile_archivers()
-          .finally([this] {
-              if (_gate.is_closed()) {
-                  return;
-              }
-              _timer.rearm(_jitter());
-          })
-          .handle_exception([this](std::exception_ptr e) {
-              vlog(_rtclog.info, "Error in timer callback: {}", e);
-          });
+    (void)ss::with_scheduling_group(_upload_sg, [this] {
```

Comment:
i think that this should be the other way around, i.e. first enter the gate, then access scheduling group

Reply:
i think that before it might happen that object was destroyed before accessing `_upload_sg`. When gate is entered first, that is impossible.